### PR TITLE
Reconcile application_type validation with PK-Sim's canonical set

### PR DESCRIPTION
**Pull request by Felixmil**
Closes #171

Reconcile `application_type` validation with PK-Sim's canonical set so the package accepts exactly the four application-type names PK-Sim itself resolves through `ApplicationTypes.ByName()` (`Oral`, `Intravenous`, `IntravenousBolus`, `UserDefined`), and gate the target-organ / target-compartment inputs to `UserDefined`, matching PK-Sim's domain rule where those fields are only meaningful for a user-defined application. The old nine-value list both rejected values PK-Sim accepts and accepted seven values (`IntravenousInfusion`, `Intramuscular`, `Subcutaneous`, `Dermal`, `Rectal`, `Inhalation`, `Intraperitoneal`) that PK-Sim has never known, and it omitted `UserDefined`. This is a breaking change to the accepted value set.

## Enum reconciliation

`PKSIM_APPLICATION_TYPES` in `R/Protocol.R` is now exactly the four canonical entries in canonical order, each with a correct human-readable label: `Oral` -> "Oral", `Intravenous` -> "Intravenous", `IntravenousBolus` -> "Intravenous bolus", `UserDefined` -> "User defined". The vector's header comment now states it holds these four names. This single named vector is the source of truth: `schema_item_application_types()` and `human_application_type()` read from it unchanged, so the accepted set and the labels propagate automatically to both factory functions and both `application_type` R6 setters (`Protocol$application_type`, `SchemaItem$application_type`) with no logic change to any of them. `human_application_type()` keeps its raw-value fallback, so an unknown value on a loaded snapshot still round-trips its own name.

## Target-field gate

`create_schema_item()` and `create_protocol()` (Simple-Protocol branch) now reject `target_organ` and/or `target_compartment` when `application_type` is not `"UserDefined"`, raising an early `cli::cli_abort()` (before any object is built) with a message that is byte-identical between the two factories. In `create_protocol()` the predicate is additionally guarded by `!is.null(application_type)`, since a Simple protocol may legitimately carry a target field with no explicit application type. The gate is intentionally confined to the two authoring factories; it is not added to the R6 setters or the load path, so externally authored snapshots that carry a target field on a non-`UserDefined` application still round-trip byte-faithfully.

## Docs and vignette

The roxygen `@param application_type` lists in `create_schema_item()` and `create_protocol()` now name the four canonical values, and their `target_organ` / `target_compartment` params note the `UserDefined` gating. The `create_schema_item()` example that paired target fields with `IntravenousBolus` now uses `UserDefined`, so it stays runnable under the gate. The `vignettes/creating-building-blocks.Rmd` sentence that hard-coded the nine values now lists the canonical four. `man/create_schema_item.Rd` and `man/create_protocol.Rd` were regenerated with `devtools::document()` (no `NAMESPACE` change).

## NEWS

One single-line breaking-change bullet was added to the development-version section, leading with `create_schema_item()` and closing `(#171)`, stating that `application_type` is now validated against PK-Sim's canonical four values, that `UserDefined` is newly accepted, that the previously listed non-canonical values are removed, and that `target_organ` / `target_compartment` are now accepted only with `UserDefined`.

## Tests and snapshots

`test-create_schema_item.R` gained an accepted-set test enumerating the four canonical values, a gate test rejecting target fields on non-`UserDefined` types (via `expect_snapshot(error = TRUE)`), and a removed-value (`Subcutaneous`) rejection; the "carries optional fields through" test now uses `UserDefined`, and the passthrough-parameter test moved off the removed `IntravenousInfusion` to canonical `Intravenous`. `test-create_protocol.R` gained the mirrored accepted-`UserDefined`-with-targets test, the gate-rejection test, a null-type-with-target acceptance test, and a removed-value rejection. `test-Protocol.R`'s "handles different types" test dropped the removed `IntravenousInfusion` / `Subcutaneous` cases and now asserts the `Intravenous`, `IntravenousBolus`, and `UserDefined` labels, keeping the unknown-value fallback and length-1-list cases. Snapshots were re-recorded by running the affected files and reviewing the diff before accepting: `create_schema_item.md`, `create_protocol.md`, `SchemaItem.md`, and `Protocol.md` changed (the "Valid values" line shrank to the four canonical values, and the two gate-error snapshots were added); `value_specs.md` did not change.

## Example

``` r
library(osp.snapshots)
#> 
#> Attaching package: 'osp.snapshots'
#> The following object is masked from 'package:stats':
#> 
#>     time
#> The following object is masked from 'package:base':
#> 
#>     range

# `UserDefined` is newly accepted, and target_organ / target_compartment
# may be supplied with it.
item <- create_schema_item(
  name = "User defined",
  application_type = "UserDefined",
  target_organ = "VenousBlood",
  target_compartment = "Plasma"
)
item$application_type
#> [1] "UserDefined"
item$target_organ
#> [1] "VenousBlood"
item$target_compartment
#> [1] "Plasma"

# A previously accepted non-canonical value is now rejected.
create_schema_item(name = "Item", application_type = "Subcutaneous")
#> Error in `create_schema_item()`:
#> ! `application_type` must be one of the canonical PK-Sim application
#>   types.
#> ✖ Got "Subcutaneous".
#> ℹ Valid values: "Oral", "Intravenous", "IntravenousBolus", and "UserDefined".

# target_organ / target_compartment are gated to `UserDefined`.
create_schema_item(
  name = "Item",
  application_type = "Oral",
  target_organ = "Liver"
)
#> Error in `create_schema_item()`:
#> ! `target_organ` and `target_compartment` are only valid when
#>   `application_type` is "UserDefined".
#> ✖ Got `application_type` = "Oral".
#> ℹ Remove the target field(s), or set `application_type` to "UserDefined".
```

## Verification

All R runs used `NOT_CRAN=true`. Ran the four affected test files individually to re-record snapshots (`testthat::test_file()` after `devtools::load_all()`), reviewed the `_snaps` diff, and accepted only `create_schema_item.md`, `create_protocol.md`, `SchemaItem.md`, `Protocol.md`. Ran the full suite with `devtools::test()`: 3244 pass, 0 fail, 1 skip. Regenerated docs with `devtools::document()` (only `create_schema_item.Rd` and `create_protocol.Rd` changed; `NAMESPACE` unchanged). Confirmed `schema_item_application_types()` returns the canonical four and `human_application_type()` returns "Oral", "Intravenous", "Intravenous bolus", "User defined" per value plus the raw-value fallback for an unknown value and the length-1-list case. Ran `devtools::run_examples()` with no example errors, confirming the `UserDefined` example clears the gate.

# QA: Approved

The PR reconciles `application_type` to PK-Sim's canonical four values and gates `target_organ`/`target_compartment` to `UserDefined` at both factories exactly as the spec and plan require; every FR and AC is satisfied, the full suite is green (3244 pass, 0 fail, 1 skip under `NOT_CRAN=true`, matching the build claim, re-run independently), and there is no scope creep.

<details>
<summary>Full QA report</summary>

## Summary of verification method

Reviewed the diff (`gh pr diff 180`), read the changed source directly in the worktree, and independently re-ran: the affected test files, the full suite, `devtools::document()` (to confirm no `man/` drift), `devtools::run_examples()` for the two changed man pages, and targeted `load_all()` probes for the enum, labels, setters, and load-path round-trip.

## Functional requirements

- **FR-1 (four-entry vector + comment)**: Pass. `R/Protocol.R:530-535` holds exactly `Oral`, `Intravenous`, `IntravenousBolus`, `UserDefined` in canonical order with correct labels. The header comment (lines 520-529) now states "knows exactly these four names" and no longer lists removed values.
- **FR-2 (`schema_item_application_types()` unchanged, four names)**: Pass. No change to the accessor body; `load_all()` confirms it returns `c("Oral","Intravenous","IntravenousBolus","UserDefined")`.
- **FR-3 (`create_schema_item()` enum validation)**: Pass. The existing three-line `cli::cli_abort()` at `R/create_schema_item.R:90-96` is intact; the valid-value list now renders the four canonical names (verified in the re-recorded `create_schema_item.md`, `SchemaItem.md` snapshots).
- **FR-4 (`create_schema_item()` target gate)**: Pass. Gate at `R/create_schema_item.R:97-106` is a presence check that runs after the enum validation and before `data <- list(...)` (line 162), so a rejected call builds no object. `UserDefined` writes both fields as before; a non-`UserDefined` type with no target fields is unchanged (verified `Oral` with no targets succeeds).
- **FR-5 (`create_protocol()` Simple-branch gate + shared enum)**: Pass. Gate at `R/create_protocol.R:202-212`, inside the `else` branch, below the `schemas` mutual-exclusivity check (lines 142-163), after the enum check, and before the `TargetOrgan`/`TargetCompartment` writes (lines 213-218). The `!is.null(application_type)` guard is present so a null-type Simple protocol may still carry targets (verified: `create_protocol(name="P", target_organ="Liver")` succeeds and writes `TargetOrgan`).
- **FR-6 (both `application_type` setters)**: Pass. No setter code change. Verified `Protocol$application_type` accepts `Intravenous`, clears on `NULL`, and rejects `Subcutaneous`. `R/SchemaItem.R` source is untouched.
- **FR-7 (`human_application_type()` labels, `to_df`)**: Pass. Function body unchanged (`R/Protocol.R:541-548`); only the label vector changed. Per-value labels verified: `Oral`/`Intravenous`/`Intravenous bolus`/`User defined`, raw fallback for `FutureRouteOfAdministration`, and list-input (`list("IntravenousBolus")`) resolves.
- **FR-8 (roxygen, example, field doc, `man/`, vignette)**: Pass. Both `@param application_type` lists show exactly the four values; the `create_schema_item()` second example switched to `UserDefined` and runs under the gate (`run_examples()` clean); `target_organ`/`target_compartment` params note the gating; `R/SchemaItem.R` field-doc examples were correctly left (already canonical); `vignettes/creating-building-blocks.Rmd:297` now lists the four values (plan AM-1); `man/create_schema_item.Rd` and `man/create_protocol.Rd` regenerated, and a fresh `devtools::document()` produced no diff.
- **FR-9 (NEWS breaking-change bullet)**: Pass. Single-line bullet at `NEWS.md:6`, leading with `create_schema_item()`, `(#171)` before the period, present tense, backticked functions, breaking-change framing, no intra-cycle narration. Placement is alphabetically correct within the already-alphabetised dev list (after the `create_schema_item()` #168 bullet, before `create_snapshot()`) and ahead of the `fraction_unbound()` #156 bullet as required. Note: the plan's premise that the dev section held "only the #156 bullet" was stale (it now has several bullets), but the ordering intent is met.

## Acceptance criteria

- **AC-1 (set reconciled)**: Pass. `schema_item_application_types()` returns exactly the four; the seven non-canonical values are rejected by both factories and the setter (verified via snapshots and a live setter probe rejecting `Subcutaneous`).
- **AC-2 (UserDefined interaction defined)**: Pass. Accepted `UserDefined`-with-targets paths tested in both `test-create_schema_item.R` (the "carries optional fields through" test now uses `UserDefined`) and `test-create_protocol.R` (new acceptance test). Rejected paths captured via `expect_snapshot(error = TRUE)` in both files (Oral+`target_organ`, IntravenousBolus+`target_compartment`). Gate runs before the object/`data` is built; the `create_protocol()` null-type guard is preserved.
- **AC-3 (accepted set + rejected value)**: Pass. `test-create_schema_item.R` enumerates all four canonical values through the factory and asserts each round-trips; a removed value (`Subcutaneous`) and `NotARealType` are rejected via snapshot. `test-create_protocol.R` mirrors the removed-value rejection.
- **AC-4 (labels correct)**: Pass. Labels verified per value plus raw fallback; `test-Protocol.R` asserts `Intravenous`, `Intravenous bolus`, and `User defined`, keeping the `FutureRouteOfAdministration` fallback and length-1-list cases. `human_application_type()` body was not altered.
- **AC-5 (round-trip preserved)**: Pass. Setters and load path are ungated (no `R/SchemaItem.R` source change, no mapper/serializer/load-path files touched). Verified directly: `SchemaItem$new(list(ApplicationType="IntravenousBolus", TargetOrgan="Liver"))` round-trips both fields. Fixture JSON under `tests/testthat/data/` is untouched and still loads.
- **AC-6 (docs and examples runnable)**: Pass. Fresh `devtools::document()` yields no diff (regenerated, not hand-edited); `@param` lists show exactly the four; `run_examples()` for both changed man pages runs clean, so the `UserDefined` example clears the gate.
- **AC-7 (suite green)**: Pass. Independently re-ran `devtools::test()` under `NOT_CRAN=true`: 3244 pass, 0 fail, 1 skip, matching the build claim. The 23 warnings are pre-existing and unrelated to this change.

## Repo conventions and scope

- NEWS: one single-line bullet, leading `create_schema_item()`, `(#171)` before the period, alphabetised, no journey narration. Pass.
- Commit message: imperative summary, grouped per-file bullets, no `Co-Authored-By` trailer. Pass.
- Gate message: byte-identical between the two factories (only source indentation differs). Pass.
- Snapshot changes confined to `create_schema_item.md`, `create_protocol.md`, `SchemaItem.md`, `Protocol.md`; `value_specs.md` unchanged. Pass.
- Scope: only `R/Protocol.R`, `R/create_protocol.R`, `R/create_schema_item.R`, the two regenerated man pages, NEWS, the vignette, and the mapped tests/snapshots changed. No R6 setter, mapper, serializer, or load-path edit leaked in. Pass.

## Adversarial and edge-case findings

- Partial-object edge case: In `create_protocol()` the gate fires after `data$ApplicationType`/`data$DosingInterval` are appended to the local `data`, but before the target-field writes and before any return; because `cli_abort()` throws, no partially built object is ever observable. This matches the plan's stated placement (after the enum check, before the target writes) and the spec's intent (no partial object returned). Not a defect.
- Null-type Simple protocol with a target field: the `!is.null(application_type)` guard is the deliberate, tested difference from the schema-item gate; dropping it would break the documented edge case. Present and covered.
- Externally authored non-`UserDefined` snapshot carrying `TargetOrgan`: round-trips unchanged through the ungated setter. Verified live.
- No correctness bug, missed step, or scope creep found.

</details>

## Summary by CodeRabbit

* **Bug Fixes**
  * Enforced stricter `application_type` validation, accepting only `Oral`, `Intravenous`, `IntravenousBolus`, and `UserDefined` (other routes are now rejected).
  * Restricted `target_organ` and `target_compartment` so they’re only allowed when `application_type = "UserDefined"`, with clearer error messages when misused.
* **Documentation**
  * Updated help text, examples, changelog/vignettes, and snapshots to reflect the supported application types and the conditional `target_*` field rules.

### CODE REVIEW

File: R/create_protocol.R [hunk 1]
```diff
@@ -199,6 +199,17 @@ create_protocol <- function(
     if (!is.null(dosing_interval)) {
       data$DosingInterval <- dosing_interval
     }
+    if (
+      !is.null(application_type) &&
+        application_type != "UserDefined" &&
+        (!is.null(target_organ) || !is.null(target_compartment))
+    ) {
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply R/create_protocol.R:205]: Fixed in 44b76ea. You are right: the null-`application_type` case slipped through the gate, which contradicted the documented "target fields are only valid for `UserDefined`" contract. The Simple-branch gate now fires for the null case too:

```r
if (
  (is.null(application_type) || application_type != "UserDefined") &&
    (!is.null(target_organ) || !is.null(target_compartment))
) {
  cli::cli_abort(...)
}
```

The error message is NULL-tolerant (it reports `Got application_type = NULL.` when the type is omitted). A typeless Simple protocol with no target fields is still accepted; only the target-carrying case is now rejected.

File: tests/testthat/test-create_protocol.R [hunk 2]
```diff
@@ -234,6 +234,49 @@ test_that("create_protocol validates required arguments", {
...
+    )
+  )
+})
+
+test_that("create_protocol allows target fields when application_type is NULL", {
+  protocol <- create_protocol(name = "P", target_organ = "Liver")
+  expect_s3_class(protocol, "Protocol")
+  expect_null(protocol$application_type)
+  expect_equal(protocol$data$TargetOrgan, "Liver")
 })
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply tests/testthat/test-create_protocol.R]: Fixed in 44b76ea. The test that asserted `create_protocol(name = "P", target_organ = "Liver")` serialized `TargetOrgan` was replaced by one asserting it now errors (captured with `expect_snapshot(error = TRUE)`), matching the tightened gate. The `UserDefined`-with-targets acceptance test and the non-`UserDefined` rejection test are unchanged.

File: NEWS.md [hunk 3]
```diff
@@ -4,6 +4,7 @@
...
 - `create_schema_item()` gains promoted `dose` (with `dose_unit`, default `"mg"`) and `start_time` (with `start_time_unit`, default `"h"`) arguments, mirroring `create_output_interval()` and `create_event_selection()`; the dose is written as a single `InputDose` parameter whose unit selects the dose family, and supplying a promoted argument together with the matching `parameters` entry is an error (#168).
+- `create_schema_item()` (and `create_protocol()`, `Protocol$application_type`, and `SchemaItem$application_type`) now validate `application_type` against PK-Sim's canonical four values (`Oral`, `Intravenous`, `IntravenousBolus`, `UserDefined`), newly accepting `UserDefined` and no longer accepting the previously listed non-canonical values (`IntravenousInfusion`, `Intramuscular`, `Subcutaneous`, `Dermal`, `Rectal`, `Inhalation`, `Intraperitoneal`); `target_organ` and `target_compartment` are now accepted only when `application_type` is `UserDefined` (#171).
```
[the remark this answers is not in the document]
Comment (Felixmil) [reply NEWS.md:7]: Skipping this one: the premise no longer holds after commit 44b76ea. Simple protocols without an explicit `application_type` can no longer carry `target_organ` / `target_compartment`; that null-type case now errors, same as any non-`UserDefined` type. So the NEWS wording ("accepted only when `application_type` is `UserDefined`") is accurate as written, and adding a "Simple-protocol exception" would make the changelog describe behavior the code no longer has. This is consistent with your confirmation on the `R/create_protocol.R:205` thread that the guard "correctly rejects target fields when `application_type` is omitted." Leaving the bullet unchanged.
